stdlib/src/locale.rs

```diff
@@ -67,76 +127,144 @@ mod _locale {
         )
     }
 
+    #[cfg(any(
+        unix,
+        windows,
+        not(any(target_os = "ios", target_os = "android", target_arch = "wasm32"))
+    ))]
     #[pyfunction]
     fn strcoll(string1: PyStrRef, string2: PyStrRef, vm: &VirtualMachine) -> PyResult {
         let cstr1 = CString::new(string1.as_str()).map_err(|e| e.to_pyexception(vm))?;
         let cstr2 = CString::new(string2.as_str()).map_err(|e| e.to_pyexception(vm))?;
         Ok(vm.new_pyobj(unsafe { libc::strcoll(cstr1.as_ptr(), cstr2.as_ptr()) }))
     }
 
+    #[cfg(any(
+        unix,
+        windows,
+        not(any(target_os = "ios", target_os = "android", target_arch = "wasm32"))
+    ))]
     #[pyfunction]
     fn strxfrm(string: PyStrRef, vm: &VirtualMachine) -> PyResult {
         // https://github.com/python/cpython/blob/eaae563b6878aa050b4ad406b67728b6b066220e/Modules/_localemodule.c#L390-L442
         let n1 = string.byte_len() + 1;
-        let mut buff: Vec<u8> = vec![0; n1];
+        let mut buff = vec![0u8; n1];
 
         let cstr = CString::new(string.as_str()).map_err(|e| e.to_pyexception(vm))?;
         let n2 = unsafe { libc::strxfrm(buff.as_mut_ptr() as _, cstr.as_ptr(), n1) };
-        buff.truncate(n2);
+        buff = vec![0u8; n2 + 1];
+        unsafe {
+            libc::strxfrm(buff.as_mut_ptr() as _, cstr.as_ptr(), n2 + 1);
+        }
         Ok(vm.new_pyobj(String::from_utf8(buff).expect("strxfrm returned invalid utf-8 string")))
     }
 
-    #[pyfunction]
-    fn localeconv(vm: &VirtualMachine) -> PyResult<PyDictRef> {
+    #[cfg(all(windows, not(target_arch = "wasm32")))]
+    #[pyfunction(name = "localeconv")]
+    fn _localeconv_w(vm: &VirtualMachine) -> PyResult<PyDictRef> {
```

Comment:
`localeconv_w` and `localeconv_u` seems same function except for the i. is it? in that case, can we share a single `localeconv` for both case?